databuilder/dsl.py

```diff
@@ -367,6 +408,49 @@ def convert_to_days(self):
     def convert_to_weeks(self):
         return self._convert("weeks")
 
+    @staticmethod
+    def _delta_in_days(datedelta):
+        """
+        Convert a DateSeltaSeries representing the difference between two dates to an
+        IntSeries representing days.
+        """
+        if isinstance(datedelta, DateDeltaSeries) and isinstance(
+            datedelta.value, DateDifference
+        ):
+            return datedelta.convert_to_days()
+        return datedelta
+
+    def __add__(
+        self, other: DateDeltaSeries | DateSeries | int
+    ) -> DateDeltaSeries | DateSeries:
+        # Adding a DateSeries to a DateDeltaSeries should use DateSeries.__add__ to
+        # return a new DateSeries
+        if isinstance(other, DateSeries):
+            return other + self
+        return DateDeltaSeries(
+            DateDeltaAddition(self._delta_in_days(self), self._delta_in_days(other))
+        )
+
+    def __radd__(self, other: DateSeries | int) -> DateDeltaSeries | DateSeries:
+        return self + other
+
+    def __sub__(self, other: DateDeltaSeries | int) -> DateDeltaSeries:
+        return DateDeltaSeries(
+            DateDeltaSubtraction(self._delta_in_days(self), self._delta_in_days(other))
+        )
+
+    def __rsub__(
+        self, other: str | DateDeltaSeries | int
+    ) -> DateSeries | DateDeltaSeries:
```

Comment:
I wondered whether `other` could be a `DateSeries`.   But I think I've realised it couldn't be, because that would be handled by `DateSeries.__sub__`.  Could you add a note explaining that?

Reply:
Aaah, this comment has finally made me figure out what mypy was complaining about with the DateSeries __radd__.  I think I can fix that now :)